core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBomScanner.java

```diff
@@ -137,9 +153,19 @@ public void removedBundle(Bundle bundle, BundleEvent bundleEvent, Iterable<? ext
             }
         }
 
+        private void removeFromCatalog(CatalogItem<?, ?> item) {
+            try {
+                getManagementContext().getCatalog().deleteCatalogItem(item.getSymbolicName(), item.getVersion());
+            } catch (Exception e) {
+                LOG.warn("Failed to remove {} {} from catalog as {}", new String[] {
```

Comment:
`propagateIfFatal` got lost in the rebase. Better insert it as the first expression after the catch.